tests/unit_tests/dataframe_test.py

```diff
@@ -53,3 +58,44 @@ def test_js_max_int(app_context: None) -> None:
         {"a": 1, "b": "1239162456494753670", "c": "c1"},
         {"a": 2, "b": 100, "c": "c2"},
     ]
+
+
+@pytest.mark.parametrize(
+    "inpt, expected",
+    [
+        pytest.param(
+            [
+                (datetime.strptime("1677-09-22 00:12:43", "%Y-%m-%d %H:%M:%S"), 1),
+                (datetime.strptime("2262-04-11 23:47:17", "%Y-%m-%d %H:%M:%S"), 2),
+            ],
+            [
+                {"a": datetime.strptime("1677-09-22 00:12:43", "%Y-%m-%d %H:%M:%S"), "b": 1},
+                {"a": datetime.strptime("2262-04-11 23:47:17", "%Y-%m-%d %H:%M:%S"), "b": 2},
+            ],
+            id="timestamp conversion fail"
+        ),
+        pytest.param(
+            [
+                (datetime.strptime("1677-09-22 00:12:44", "%Y-%m-%d %H:%M:%S"), 1),
+                (datetime.strptime("2262-04-11 23:47:16", "%Y-%m-%d %H:%M:%S"), 2)
+            ],
+            [
+                {"a": Timestamp("1677-09-22 00:12:44"), "b": 1},
+                {"a": Timestamp("2262-04-11 23:47:16"), "b": 2}
+            ],
+            id="timestamp conversion success"
+        )
+    ]
+)
+def test_max_pandas_timestamp(inpt, expected) -> None:
+    from superset.db_engine_specs import BaseEngineSpec
+    from superset.result_set import SupersetResultSet
+
+    cursor_descr: DbapiDescription = [
+        ("a", "datetime", None, None, None, None, False),
+        ("b", "int", None, None, None, None, False),
+    ]
+    results = SupersetResultSet(inpt, cursor_descr, BaseEngineSpec)
```

Comment:
nit: this feels more pythonic:
```suggestion
@pytest.mark.parametrize(
    "input_, expected",
    [
        pytest.param(
            [
                (datetime.strptime("1677-09-22 00:12:43", "%Y-%m-%d %H:%M:%S"), 1),
                (datetime.strptime("2262-04-11 23:47:17", "%Y-%m-%d %H:%M:%S"), 2),
            ],
            [
                {"a": datetime.strptime("1677-09-22 00:12:43", "%Y-%m-%d %H:%M:%S"), "b": 1},
                {"a": datetime.strptime("2262-04-11 23:47:17", "%Y-%m-%d %H:%M:%S"), "b": 2},
            ],
            id="timestamp conversion fail"
        ),
        pytest.param(
            [
                (datetime.strptime("1677-09-22 00:12:44", "%Y-%m-%d %H:%M:%S"), 1),
                (datetime.strptime("2262-04-11 23:47:16", "%Y-%m-%d %H:%M:%S"), 2)
            ],
            [
                {"a": Timestamp("1677-09-22 00:12:44"), "b": 1},
                {"a": Timestamp("2262-04-11 23:47:16"), "b": 2}
            ],
            id="timestamp conversion success"
        )
    ]
)
def test_max_pandas_timestamp(input_, expected) -> None:
    from superset.db_engine_specs import BaseEngineSpec
    from superset.result_set import SupersetResultSet

    cursor_descr: DbapiDescription = [
        ("a", "datetime", None, None, None, None, False),
        ("b", "int", None, None, None, None, False),
    ]
    results = SupersetResultSet(input_, cursor_descr, BaseEngineSpec)
```

Reply:
Thanks for reviewing! Should be changed now